### PR TITLE
Create initial frontend

### PR DESCRIPTION
**Pull request by cjdenio**
Creates a basic Next.js frontend! Also switches to the Gin web framework and separates dev/prod Dockerfiles.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,33 @@
...
+RUN yarn build
+## BUILD FRONTEND ##
+
+## BUILD BACKEND ##
+FROM golang:1.16-alpine
+
+WORKDIR /usr/src/app
+
+COPY . .
+COPY --from=0 /usr/src/app/out/ ./frontend/out
```
Comment (anirudhb) [inline Dockerfile]: Name each stage so that it doesn't get out of sync if new stages are added

**Review by anirudhb — CHANGES_REQUESTED** [review]
lgtm, nit
new: please update the README to remove dokkud_test

**Comment by jasonappah** [comment]
One other thing which might be out of the scope of this PR: How will running the frontend in dev work? AFAICS the the dev Dockerfile doesn't build the frontend at all.

**Comment by cjdenio** [comment]
@jasonappah You'll just need to run `yarn dev` in the frontend dir; it's not realistic to run it in Docker.

Once the frontend needs to talk with the API, we should have an `API_ROOT` frontend env var that changes based on dev/prod

**Comment by jasonappah** [comment]
Ah makes sense, everything else LGTM

**Review by anirudhb — APPROVED** [review]
lgtm -- consider if TypeScript would be beneficial here

**Comment by cjdenio** [comment]
TypeScript would be awesome, I was just too lazy to set it up this time lol

**Pull request merged**